t/03-testapi.t

```diff
@@ -576,30 +578,144 @@ subtest 'wait_still_screen & assert_still_screen' => sub {
         'assert_still_screen forwards arguments to wait_still_screen');
 };
 
-subtest 'set console tty and other args' => sub {
-    # ensure previously emitted commands are cleared
-    $cmds = ();
+subtest 'testapi::console() => backend::console_proxy => backend::baseclass::proxy_console_call()' => sub {
+    use backend::baseclass;
+    $cmds = [];
+    my $wrapped_results = [];
+    my $check_args      = [];
 
-    console->set_tty(4);
-    console->set_args(tty => 5, foo => 'bar');
+    my $console = Test::MockObject->new();
+    $console->mock('ret_array', sub {
+            my @array = qw(a b c d );
+            return @array;
+    });
+    $console->mock('ret_array_empty', sub {
+            my @a;
+            return @a;
+    });
+    $console->mock('ret_scalar', sub {
+            my $s = "a";
+            return $s;
+    });
+    $console->mock('ret_undef', sub {
+            return undef;
+    });
+    my %h = (x => 'a', y => 'b', z => 'c');
+    $console->mock('ret_hash', sub {
+            return %h;
+    });
+    $console->mock('ret_hash_ref', sub {
+            return {x => 'a', y => 'b', z => 'c'};
+    });
+    $console->mock('ret_hash_ref_empty', sub {
+            return {};
+    });
+    $console->mock('ret_array_ref', sub {
+            return [qw(a b c d)];
+    });
+    $console->mock('ret_array_ref_empty', sub {
+            return [];
+    });
+    $console->mock('ret_list', sub {
+            return qw(a b c d);
+    });
+    $console->mock('ret_list_empty', sub {
+            return;
+    });
+    $console->mock('ret_die', sub {
+            die("!!Urgs!!");
+    });
+    $console->mock('check_args', sub {
+            my ($self, @args) = @_;
+            is_deeply(\@args, $check_args, 'Got expected (' . join(',', @_) . ') arguments');
+    });
 
-    is_deeply(
-        $cmds,
-        [
+    my $baseclass      = backend::baseclass->new();
+    my $mock_baseclass = Test::MockModule->new('backend::baseclass');
+    $mock_baseclass->redefine('console', sub {
+            return $console;
+    });
```

Comment:
```suggestion
    $mock_baseclass->redefine(console => sub { $console; });
```

Maybe some of the other redefines can also be written in a more compact way.

Reply:
done